### PR TITLE
Fix `/room/.../event/...` to return the *original* event after any edits

### PR DESCRIPTION
**Pull request by richvdh**
This is what the MSC (now) requires. Fixes https://github.com/matrix-org/synapse/issues/10310.

### CODE REVIEW

File: synapse/events/utils.py [hunk 1]
```diff
@@ -490,9 +494,10 @@ def _inject_bundled_aggregations(
             serialized_aggregations[RelationTypes.REFERENCE] = aggregations.references
 
         if aggregations.replace:
-            # If there is an edit, apply it to the event.
+            # If there is an edit, optionally apply it to the event.
             edit = aggregations.replace
-            self._apply_edit(event, serialized_event, edit)
+            if apply_edits:
```
Comment (clokep) [inline synapse/events/utils.py:499]: This is fine, but I think I would have combined it into the if-statement above.

Comment (richvdh) [reply synapse/events/utils.py:499]: ... but then the `serialized_aggregations` update would be missed?

Comment (clokep) [reply synapse/events/utils.py:499]: Ah, I misread the indentation, yeah this looks good! 👍

**Comment by clokep** [comment]
I know in the `/relations` endpoint we have some code that is vaguely related to this (not applying aggregations to get the "original event"):

https://github.com/matrix-org/synapse/blob/772bad25628959b4e919ea0b57c0c6be112143cb/synapse/handlers/relations.py#L153-L157

Should we be not applying edits there instead?

**Comment by richvdh** [comment]
> Should we be not applying edits there instead?

since `bundle_aggregations=None`, it looks like we won't have any edits to apply?

**Comment by clokep** [comment]
> > Should we be not applying edits there instead?
> 
> since `bundle_aggregations=None`, it looks like we won't have any edits to apply?

Right. I got myself confused; this is unrelated. 👍

**Review by clokep — APPROVED** [review]
I think this looks good! 👍

**Pull request merged**